### PR TITLE
Switch CI to Github Actions

### PR DESCRIPTION
**Pull request by glandium**
Fixes: #910

### CODE REVIEW

File: .github/actions/rust-toolchain/action.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    required: false
+  target:
+    description: Target triple to install for this toolchain
+    required: false
+  components:
+    description: Space-separated list of components to be additionally installed for a new toolchain
+    required: false
+
+runs:
+  using: composite
```
Comment (mitchhentges) [inline .github/actions/rust-toolchain/action.yml:16]: Should this be `composite`? Only a single step is run :thinking:

File: .github/actions/rust-toolchain/action.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  steps:
+    - shell: bash
+      run: |
+        set -x
```
Comment (mitchhentges) [inline .github/actions/rust-toolchain/action.yml:20]: Nice

File: .github/actions/rust-toolchain/action.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        rustup override set ${{ inputs.toolchain || 'stable' }}
+        ${{ inputs.target && format('rustup target add {0}', inputs.target) }}
+        ${{ inputs.components && format('rustup component add {0}', inputs.components) }}
+        cargo --version
+        rustc --version
```
Comment (mitchhentges) [inline .github/actions/rust-toolchain/action.yml:25]: Smart

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,40 +1,178 @@
...
+    name: release
+    runs-on: ubuntu-latest
+    needs: [build, lint, test]
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v1
+
+      - name: Get artifacts
+        uses: actions/download-artifact@v2
```
Comment (mitchhentges) [inline .github/workflows/ci.yml:163]: Where does `download-artifact` come from?

**Review by mitchhentges — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (glandium) [reply .github/workflows/ci.yml:163]: https://github.com/actions/download-artifact

File: .github/actions/rust-toolchain/action.yml (hunk 1 above)
Comment (glandium) [reply .github/actions/rust-toolchain/action.yml:16]: See https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-composite-run-steps-actions
The alternatives are docker or node.

**Pull request merged**